frontend/app/routes/_public+/intake+/$id+/communication-preference.tsx

```diff
@@ -0,0 +1,138 @@
+
+import { Button, ButtonLink } from '~/components/buttons';
+import { InputField } from '~/components/input-field';
+import { InputLegend } from '~/components/input-legend';
+import { InputRadio } from '~/components/input-radio';
+import { InputRadios } from '~/components/input-radios';
+import { getLookupService } from '~/services/lookup-service.server';
+import { getRaoidcService } from '~/services/raoidc-service.server';
+import { getSessionService } from '~/services/session-service.server';
+import { getNameByLanguage, getTypedI18nNamespaces } from '~/utils/locale-utils';
+import type { RouteHandleData } from '~/utils/route-utils';
+
+const i18nNamespaces = getTypedI18nNamespaces('communication-preference');
+
+export const handle = {
+  breadcrumbs: [
+    // prettier-ignore
+    { labelI18nKey: 'communication-preference:breadcrumbs.preferred-communication' },
+  ],
+  i18nNamespaces,
+  pageIdentifier: 'CDCP-00XX',
+  pageTitleI18nKey: 'communication-preference:page-title',
+} as const satisfies RouteHandleData;
+
+export async function loader({ request }: LoaderFunctionArgs) {
+  const raoidcService = await getRaoidcService();
+  await raoidcService.handleSessionValidation(request);
+
+  const preferredLanguages = await getLookupService().getAllPreferredLanguages();
+  const preferredCommunicationMethods = await getLookupService().getAllPreferredCommunicationMethods();
+  return json({ preferredLanguages, preferredCommunicationMethods });
+}
+
+export async function action({ request }: ActionFunctionArgs) {
+  const raoidcService = await getRaoidcService();
+  await raoidcService.handleSessionValidation(request);
+
+  const formDataSchema = z.object({
+    preferredLanguage: z.enum(['en', 'fr']),
+  });
+  const formData = Object.fromEntries(await request.formData());
+  const parsedDataResult = formDataSchema.safeParse(formData);
+  if (!parsedDataResult.success) {
+    return json({
+      errors: parsedDataResult.error.format(),
+      formData: formData as Partial<z.infer<typeof formDataSchema>>,
+    });
+  }
+
+  const sessionService = await getSessionService();
+  const session = await sessionService.getSession(request);
+  session.set('newPreferredLanguage', parsedDataResult.data.preferredLanguage);
+
+  return redirect('/personal-information/preferred-language/confirm', {
+    headers: {
+      'Set-Cookie': await sessionService.commitSession(session),
+    },
+  });
+}
+
+export default function CommunicationPreferencePage() {
+  const { preferredLanguages, preferredCommunicationMethods } = useLoaderData<typeof loader>();
+  const { i18n, t } = useTranslation(i18nNamespaces);
+  const [isDigitalMethodChecked, setDigitalMethodChecked] = useState(false);
+
+  const digitalMethodHandler = () => {
+    setDigitalMethodChecked(true);
+  };
+
+  const nonDigitalMethodHandler = () => {
+    setDigitalMethodChecked(false);
+  };
+
+  return (
+    <>
+      <p>{t('communication-preference:note')}</p>
+      <Form method="post">
```

Comment:
Add a `noValidate` flag to the `<Form>` component here.
[see @sebastien-comeau's PR](https://github.com/DTS-STN/canadian-dental-care-plan/pull/529)

Reply:
added